osi_logicaldetectiondata.proto

```diff
@@ -0,0 +1,228 @@
+syntax = "proto2";
+
+option optimize_for = SPEED;
+
+import "osi_version.proto";
+import "osi_common.proto";
+
+package osi3;
+
+//
+// \brief Processed data from one or multiple sensors as a list of logical detections. Logical detections are derived from sensor detections in a logical model through processing steps like fusion filtering, tracking etc.
+//
+// All information is given with respect to the reference frame of the logical/virtual sensor
+// \c SensorView::mounting_position (e.g. center of rear axle of the ego car) in Cartesian coordinates.
+//
+message LogicalDetectionData
+{
+    // The interface version used by the sender (i.e. the simulation
+    // environment).
+    //
+    optional InterfaceVersion version = 1;
+
+    // Header attributes of fused detections from multiple sensors and sensor types.
+    //
+    optional LogicalDetectionDataHeader header = 2;
+
+    // logical detections consisting of transformed (and potentially fused) detections from one or multiple sensors and sensor types.
+    // Logical detections are given with respect to the reference frame of the logical/virtual sensor
+    // \c SensorView::mounting_position (e.g. center of rear axle of the ego car)
+    //
+    // \note OSI uses singular instead of plural for repeated field names.
+    //
+    repeated LogicalDetection logical_detection = 3;
+}
+
+//
+// \brief The header attributes of the list of logical detections.
+//
+message LogicalDetectionDataHeader
+{
+    // Time stamp at which the transformation and optional fusion was finished in the global synchronized time.
+    //
+    // \note See \c SensorData::timestamp and
+    // \c SensorData::last_measurement_time for detailed discussions on the
+    // semantics of time-related fields.
+    //
+    optional Timestamp logical_detection_time = 1;
+
+    // Data Qualifier expresses to what extent the content of this event can be
+    // relied on.
+    //
+    optional DataQualifier data_qualifier = 2;
+
+    // The current number of valid detections in the logical detections list.
+    //
+    // \note This value has to be set if the list contains invalid logical detections.
+    //
+    // \rules
+    // is_greater_than_or_equal_to: 0
+    // \endrules
+    //
+    optional uint32 number_of_valid_logical_detections = 3;
+
+    // The IDs of the sensors that have been fused and result in the listed logical detections.
+    //
+    repeated Identifier sensor_id = 4;
```

Comment:
By adding IDs (plural) you are always saying that it must be more than one sensor right?
So LogicalDetectionDataHeader will have at least 2 instances of sensor_id, is this the case?

depending on the answer, we may need to update it in the other messages as well.

Reply:
No, 1 sensor is possible as well, as it could just contain transformed data in cartesian.
I will change to ID(s) ok?

Reply:
Yes that is ok, please also add that 1 sensor is possible as well, as it could just contain transformed data to cartesian coordinate system as a note.

Reply:
ok, fine.

Reply:
Finetuning on that explanation has just been committed. Could you please check it?